ui/core/proto_utils/equipped_item.ts

```diff
@@ -219,23 +326,46 @@ export class EquippedItem {
 	}
 
 	withRandomSuffix(randomSuffix: ItemRandomSuffix | null): EquippedItem {
-		return new EquippedItem(this._item, this._enchant, this._gems, randomSuffix, this._reforge);
+		return new EquippedItem({
+			item: this._item,
+			enchant: this._enchant,
+			gems: this._gems,
+			randomSuffix,
+			reforge: this._reforge,
+			upgrade: this._upgrade,
+		});
 	}
 
-	getWithRandomSuffixStats() {
+	getRandomPropPoints(): number {
+		return this.item.scalingOptions[this._upgrade]?.randPropPoints || this.item.randPropPoints;
+	}
+
+	getWithDynamicStats() {
 		const item = this.item;
-		if (this._randomSuffix)
+		const scalingOptions = this._item.scalingOptions[this._upgrade];
+		item.stats = item.stats.map((stat, index) => scalingOptions.stats[index] || stat);
+		item.weaponDamageMin = scalingOptions.weaponDamageMin;
+		item.weaponDamageMax = scalingOptions.weaponDamageMax;
+
+		if (this._randomSuffix) {
+			const randomPropPoints = this.getRandomPropPoints();
```

Comment:
Might as well set `item.randPropPoints` to the new value as well for completeness.